lec01-introduction.Rmd

```diff
@@ -2,3 +2,171 @@
+
+```{r}
+read_csv('/home/joel/iris.csv') # need to dry run this with windows path names...
+```
+
+
+If we want to calculate something in Excel, we would highlight the column and then click mean... <!-- I actually don't have Excel and don't know how to do this, will check tomorrow... --> In R we first give the data a nick name where it is stored, so instead of printing out to console, these values are stored in this R session and we can access it by this name `iris`, (technically we say that we have assigned it to variable `iris`).
+
+```{r}
+iris <- read_csv('/home/joel/iris.csv')
+```
+
+Now we can simply ask to calculate the mean of the data frame `iris`, column `petal_length`:
+
+```{r}
+mean(iris$petal_length)
+```
+
+If we want to know more than just the mean, the useful thing with R, is that we don't have to calculate these things one by one, but we can actually get an overview of all the columns by asking R to provide a summary of `iris`.
+
+```{r}
+summary(iris)
+```
+
+You can already see how it will save you time to use R instead of typing these calculations separately in Excel.
+ 
+## RStudio
+Before we continue, we are going to switch from using the R console, to using an interface called RStudio. RStudio includes the R console, but also many other convenient functionalities, which makes it easier to get started and to work with R. When you launch RStudio, you will see four panels:
+- The console is the same as what we have been using so far. The advantage with using the console within RStudio is that it interact with the other panels.
+- So, if I read in the iris data we were working with previously, we can view it in the console by typing `iris`, just like before, or we can also view the `iris` variable in our environment tab, which shows it to us in a spreadsheet or table format.
+- This panel shows the files in the current directory, any plots we might make later, and also the documentation, which we accessed earlier by typing `?` in front of a function. Here, the documentation is formatted in a way that is easier to read and also provides links to the related sections.
+- Finally we have the text editor panel. This is where we can write scripts, i.e. putting several commands of code together and saving them as a text document so that they are accessible for later and so that we can execute them all at once by running the script instead of typing them in one by one.
+
+So if I create a script with some of the commands you saw earlier, I can save them all as a text file and then run the entire script by clicking the run button, which executes commands in the console. We will use this text editor to explore how to make plots in R.
+
```

Comment:
This will be for next class, but I'd suggest changing the global options with everyone to prevent saving of the workspace, etc.

Reply:
Yes, I want to change the global options when they do the software installations so that we don't have to do it everytime, but it seems like that won't happen...